backend/src/test/java/de/tostsoft/solarmonitoring/SolarSystemControllerTest.java

```diff
@@ -0,0 +1,183 @@
+import de.tostsoft.solarmonitoring.repository.UserRepository;
+import de.tostsoft.solarmonitoring.service.GrafanaService;
+import de.tostsoft.solarmonitoring.service.SolarSystemService;
+import de.tostsoft.solarmonitoring.service.UserService;
+import org.apache.commons.codec.binary.Base64;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.web.client.TestRestTemplate;
+import org.springframework.boot.web.server.LocalServerPort;
+import org.springframework.http.*;
+import org.springframework.web.client.RestTemplate;
+
+import java.nio.charset.Charset;
+import java.util.Arrays;
+
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+@SpringBootTest(classes = {SolarmonitoringApplication.class},webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+public class SolarSystemControllerTest {
+    private static final Logger LOG = LoggerFactory.getLogger(SolarSystemControllerTest.class);
+
+    @LocalServerPort
+    int randomServerPort;
+
+    @Value("${grafana.user}")
+    private String grafanaUser;
+
+    @Value("${grafana.password}")
+    private String grafanaPassword;
+
+    @Value("${proxy.grafana.target.url}")
+    private String grafanaUrl;
+    @Autowired
+    private UserService userService;
+    @Autowired
+    private GrafanaService grafanaService;
+    @Autowired
+    private InfluxConnection influxConnection;
+    @Autowired
+    private UserRepository userRepository;
+    @Autowired
+    private SolarSystemRepository solarSystemRepository;
+    @Autowired
+    private TestRestTemplate restTemplate;
+    @Autowired
+    private SolarSystemService solarSystemService;
+
+    @BeforeEach
+    public void init(){
+        cleanUPData();
+    }
+    private HttpHeaders createHeaders(){
+        return new HttpHeaders() {{
+            String auth = grafanaUser + ":" + grafanaPassword;
+            byte[] encodedAuth = Base64.encodeBase64(
+                    auth.getBytes(Charset.defaultCharset()));
+            String authHeader = "Basic " + new String(encodedAuth);
+            set("Authorization", authHeader);
+            set("Content-Type","application/json; charset=UTF-8");
+        }};
+    }
+    private void cleanUPData() {
+        String json = "";
+        var entity = new HttpEntity<String>(json, createHeaders());
+        try {
+            var list = grafanaService.getFolders();
+            for (int i = 0; list.getBody().length > i; i++) {
+                var foldersDTO = list.getBody()[i];
+                grafanaService.deleteFolder(foldersDTO.getUid());
+            }
+        }catch (Exception e){
+            System.out.println("no Folder in Database");
```

Comment:
sicher no folder in database ? was ist wenn keine verbindung zu datenbank besteht ? ist keine folder nicht eine leere liste ?